db/memtable.cc

```diff
@@ -1310,7 +1314,7 @@ bool MemTable::Get(const LookupKey& key, std::string* value,
 
   // No change to value, since we have not yet found a Put/Delete
   // Propagate corruption error
-  if (!found_final_value && merge_in_progress && !s->IsCorruption()) {
+  if (!found_final_value && merge_in_progress && s->ok()) {
```

Comment:
Should we `assert(s,ok())`? If non-ok status, then `found_final_value` should always be true right?

Reply:
It can also be merge in process, added assert on this.